CMakeModules/InternalUtils.cmake

```diff
@@ -36,6 +36,22 @@ function(arrayfire_get_cuda_cxx_flags cuda_flags)
   endif()
 endfunction()
 
+function(__af_deprecate_var var access value)
+  if(access STREQUAL "READ_ACCESS")
+    message(DEPRECATION "Variable ${var} is deprecated. Use AF_${var} instead.")
+  endif()
+endfunction()
+
+function(af_deprecate var newvar)
+  if(DEFINED ${var})
+    message(DEPRECATION "Variable ${var} is deprecated. Use ${newvar} instead.")
+    get_property(doc CACHE ${newvar} PROPERTY HELPSTRING)
```

Comment:
where is this `doc` defined ? Is it predefined ? I didn't find any reference to it in root cmake script or here.

Reply:
Its the variable where the `get_property` command stores the `HELPSTRING` for the `${newvar}` variable.